### PR TITLE
[WIP] Split content in process(), expose _bucketing_samples()

### PR DESCRIPTION
**Pull request by OscarWang114**
Update: Ready for review!

As the leaderboard in the web frontend needs an overall score for each system, I will also be saving the results from `get_overall_performance` in addition to sufficient statistics. 

Two functions are created:
1. `get_overall_statistics` for getting sufficient statistics and overall performance. It is also called by `process()` for reusability. For the web interface, the returned results are saved to the DB.
2. `get_fine_grained_statistics` exposes `_bucketing_samples` for the web interface to call on the fly, since we're not supposed to call underscore functions from outside.

Let me know if you have any concerns!

### CODE REVIEW

File: explainaboard/processors/processor.py [hunk 1]
```diff
@@ -356,6 +356,13 @@ def process(self, metadata: dict, sys_output: List[dict]) -> SysOutputInfo:
         active_features = self._complete_features(
             sys_info, sys_output, statistics=statistics
```
Comment (pfliu-nlp) [inline explainaboard/processors/processor.py:364]: it seems that we have multiple types of [`statistics`](https://github.com/neulab/ExplainaBoard/pull/162/commits/d863afd2e416a038a2cf4800cc9912f727c44c4d), do we want to make the name of this `statistics` more clear?

Comment (neubig) [reply explainaboard/processors/processor.py:364]: I think that's fixed in the other PR, which @OscarWang114 will still need to merge into this one.

Comment (OscarWang114) [reply explainaboard/processors/processor.py:364]: Thanks. Yes, I will merge them now. Also, this is still a draft so just let me know if the high-level direction sounds good, I will convert it to a formal one once the details are properly implemented.

Comment (pfliu-nlp) [reply explainaboard/processors/processor.py]: thanks! it looks good and I will be ready whenever needed to be reviewed.

Comment (OscarWang114) [reply explainaboard/processors/processor.py]: Thanks! It's ready now.

Comment (pfliu-nlp) [reply explainaboard/processors/processor.py]: Thanks will do it now.

File: explainaboard/processors/processor.py [hunk 2]
```diff
@@ -35,11 +37,19 @@ def __init__(self) -> None:
...
-        of statistics.
+    def _get_statistics_resources(
+        self, dataset_split: Dataset
+    ) -> Optional[Mapping[str, Any]]:
+        """
+        From a DataLab dataset split, get resources necessary to calculate statistics
+        """
+        return None
+
+    def _gen_external_stats(
```
Comment (pfliu-nlp) [inline explainaboard/processors/processor.py:48]: there is a `statistics` inside this function,  would it be a good idea to rename them as `external_stats` in the future?

**Review by pfliu-nlp — COMMENTED** [review]
overall it looks good

**Comment by pfliu-nlp** [comment]
Overall it looks good, and I particularly care about the concept of `statistics`, I merged this PR first and we can make other relevant discussion here: https://github.com/neulab/ExplainaBoard/issues/164

**Comment by OscarWang114** [comment]
Thanks @pfliu-nlp! I'll take a look at the discussion later today.

**Pull request merged**